File: edu/djc/fonctions.py
# Les fonctions viennent d'abord en Python, pour les charger 
# en mémoire lors de l'interprétation du script afin que les
# mots soient reconnus plus bas

def pas_d_argument():
    print("Je suis une fonction sans arguments ni valeur de retour")


def un_argument(msg):
    print("J'imprime le message passé en argument :", msg)


def multiplier_par_3(num):
    return num * 3


def multiplier_deux_valeurs(a, b):
    return a * b


# Script utilisant les fonctions définies plus haut
print("print() est une fonction pré-définie dans Python")

pas_d_argument()
un_argument("coucou")
res1 = multiplier_par_3(9)
print(res1)
res2 = multiplier_deux_valeurs(6, 7)
print(res2)

File: edu/djc/Fonctions.java
package edu.djc;
// On veut cacher le mécanisme d'exécution derrière un seul nom

/**
 * Exemples de fonctions et de méthodes sur des objets
 *
 */

public class Fonctions
{
    /**
     * Script de notre programme
     * <p>
     * main est une fonction
     * @param args 
     */
    public static void main(String[] args)
    {
        System.out.println("bonjour"); // println() est une méthode sur l'objet console (System.out)

        pasDArgument();
        unArgument("coucou");
        int result = multiplierPar3(9);
        System.out.println(result);
        int result2 = multiplierDeuxValeurs(6, 7);
        System.out.println(result2);
    }

    static void pasDArgument()
    {
        System.out.println("Je suis une fonction sans arguments ni valeur de retour");
    }

    static void unArgument(String msg)
    {
        System.out.println("J'imprime le message passé en argument : " + msg);
    }

    static int multiplierPar3(int num)
    {
        return num * 3;
    }

    static int multiplierDeuxValeurs(int a, int b)
    {
        return a*b;
    }
}